src/routes/delegates/pipes/delete-delegate-dto-validation.pipe.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpStatus, Injectable, PipeTransform } from '@nestjs/common';
+import { ValidateFunction } from 'ajv';
+import { GenericValidator } from '../../../domain/schema/generic.validator';
+import { JsonSchemaService } from '../../../domain/schema/json-schema.service';
+import { DeleteDelegateDto } from '../entities/delete-delegate.entity';
+import { deleteDelegateDtoSchema } from '../entities/schemas/delete-delegate.schema';
+
+@Injectable()
+export class DeleteDelegateDtoValidationPipe
+  implements PipeTransform<any, DeleteDelegateDto>
+{
+  private readonly isValid: ValidateFunction<DeleteDelegateDto>;
+
+  constructor(
+    private readonly genericValidator: GenericValidator,
+    private readonly jsonSchemaService: JsonSchemaService,
+  ) {
+    this.isValid = this.jsonSchemaService.getSchema(
+      'https://safe-client.safe.global/schemas/delegates/delete-delegate.json',
+      deleteDelegateDtoSchema,
+    );
+  }
+  transform(data: any): DeleteDelegateDto {
+    return this.genericValidator.validate(
+      this.isValid,
+      data,
+      HttpStatus.BAD_REQUEST,
```

Comment:
I understand the need to parameterise this but I think every route would eventually map the error to a `BAD_REQUEST`.

For now I think it's ok but the interface around validation is getting a bit convoluted (i.e. deals with schema validation _AND_ http).

Having an exception thrown and caught on the route that would map to a specific code seems to be a more scalable approach.

Reply:
Yes, I agree maybe this was introducing too much complexity into the validation logic. I've reverted the parametrization, so with this approach, we're just mapping the status code in the pipe directly: [7287b89](https://github.com/5afe/safe-client-gateway-nest/pull/271/commits/7287b895b0e15e316a3b2d81986efa8dc4e0aeb7) What do you think?